src/lib/components/Error/WidgetPropValidator.tsx

```diff
@@ -0,0 +1,36 @@
+import { WidgetProps } from 'lib/components/Widget'
+import { IntegrationError } from 'lib/errors'
+import { PropsWithChildren, useEffect, useMemo, useState } from 'react'
+
+export default function WidgetPropValidator(props: PropsWithChildren<WidgetProps>) {
+  const { jsonRpcEndpoint, provider } = props
+
+  const [providerChecked, setProviderChecked] = useState(false)
+  useEffect(() => {
+    setProviderChecked(false)
+    if (!provider && !jsonRpcEndpoint) {
+      throw new IntegrationError('This widget requires a provider or jsonRpcEndpoint.')
+    }
+    setProviderChecked(true)
+  }, [provider, jsonRpcEndpoint])
+
+  // size constraints
+  const [sizeConstraintsChecked, setSizeConstraintsChecked] = useState(false)
+  const { width } = props
+  useEffect(() => {
+    setSizeConstraintsChecked(false)
+    if (width && width < 300) {
+      throw new IntegrationError('Set widget width to at least 300px.')
+    }
+    setSizeConstraintsChecked(true)
+  }, [width])
+
+  const propsChecked = useMemo(
+    () => providerChecked && sizeConstraintsChecked,
+    [providerChecked, sizeConstraintsChecked]
+  )
+  if (propsChecked) {
```

Comment:
To reduce nesting, have this return `null` instead, and make it an adjacent DOM node

Reply:
The nesting is intentional. We don't want to render dom that's going to use these props if they are invalid.

Reply:
They will still not render, because the error should be thrown first, causing the `ErrorBoundary` to render instead.